### PR TITLE
feat: Output build_id for Android to proper locations

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

`datadog-ci` will expect a `build_id` file at `unityLibrary/symbols` alongside the LineNumberMappings.json (which is copied there) and the `.so` files that Unity generates from its build.

During build, we output the build id that Unity gives us to both `unityLibrary/symbols/build_id' (for use with `datadog-ci`) and 'unityLibrary/src/main/assets/datadog.buildId` (where it can be picked up by the Android build process and put in the application for the Android SDK to read).

This isn't 100% equivalent to the Android Gradle Plugin build_id generation, as it's not generated on each gradle build, but I believe it should work for our purposes.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests

### CODE REVIEW

File: samples/Datadog Sample/ProjectSettings/GvhProjectSettings.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <projectSetting name="Google.IOSResolver.PodfileAllowPodsInMultipleTargets" value="True" />
   <projectSetting name="Google.IOSResolver.PodfileAlwaysAddMainTarget" value="True" />
   <projectSetting name="Google.IOSResolver.PodfileEnabled" value="True" />
-  <projectSetting name="Google.IOSResolver.PodfileStaticLinkFrameworks" value="False" />
+  <projectSetting name="Google.IOSResolver.PodfileStaticLinkFrameworks" value="True" />
```
Comment (0xnm) [inline samples/Datadog Sample/ProjectSettings/GvhProjectSettings.xml:11]: Is this change related to the PR?

File: packages/Datadog.Unity/Editor/SymbolAssemblyBuildProcess.cs [hunk 2]
```diff
@@ -95,25 +98,55 @@ internal void AndroidCopyMappingFile(DatadogConfigurationOptions options, string
...
-                    var buildIdPath = Path.Join(symbolsDir, "build_id");
-                    _fileSystemProxy.WriteAllText(buildIdPath, buildGuid);
+                var buildIdPath = Path.Join(symbolsDir, "build_id");
+                _fileSystemProxy.WriteAllText(buildIdPath, buildGuid);
+
+                if (platformGroup == BuildTargetGroup.Android)
+                {
+                    // Write the build id to the Android assets directory
+                    var androidAssetsDir = Path.Join(outputPath, "unityLibrary/src/main/assets");
+                    var androidBuildIdPath = Path.Join(androidAssetsDir, "datadog.buildId");
```
Comment (0xnm) [inline packages/Datadog.Unity/Editor/SymbolAssemblyBuildProcess.cs:143]: unity builds don't have a concept of variants?

File: samples/Datadog Sample/ProjectSettings/GvhProjectSettings.xml (hunk 1 above)
Comment (fuzzybinary) [reply samples/Datadog Sample/ProjectSettings/GvhProjectSettings.xml:11]: It's not, but a generalized build change so it got looped in here.

File: packages/Datadog.Unity/Editor/SymbolAssemblyBuildProcess.cs (hunk 2 above)
Comment (fuzzybinary) [reply packages/Datadog.Unity/Editor/SymbolAssemblyBuildProcess.cs:143]: If it does, I'm not sure how to access them or add things to them. When you build the project from Unity, it does perform some incremental building, but then overwrites the Android project file with the new stuff.

The one place we *might* get in trouble is a new build from already exported project, but from what I can tell the major symbol files are built by that point and wouldn't change for the new android build.

**Review by 0xnm — APPROVED** [review]
